code/controllers/subsystem/migration.dm

```diff
@@ -283,13 +283,16 @@ SUBSYSTEM_DEF(migrants)
 		return FALSE
 	if(is_migrant_banned(player.ckey, role.name))
 		return FALSE
+	var/reason
 	if(role.allowed_races && !(prefs.pref_species.id in role.allowed_races))
 		if(!(player.triumph_ids.Find("race_all")))
-			return FALSE
+			reason += "<br>Wrong species. Your prioritized role only allows [role.allowed_races.Join(", ")]. "
```

Comment:
Having a non boolean return value in a boolean check function is bad, you should just output to the client here and early return in migrant_pref.dm